contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,275 @@
+    */
+    function modifyBlacklistType(uint256 _startTime, uint256 _endTime, bytes32 _name, uint256 _repeatPeriodTime) public withPerm(ADMIN) {
+        require(blacklists[_name].endTime != 0, "Blacklist type doesn't exist"); 
+        _validParams(_startTime, _endTime, _name, _repeatPeriodTime);
+        blacklists[_name] = BlacklistsDetails(_startTime, _endTime, _repeatPeriodTime);
+        emit ModifyBlacklistType(_startTime, _endTime, _name, _repeatPeriodTime);
+    }
+
+    /**
+    * @notice Used to delete the blacklist type
+    * @param _name name of the blacklist type
+    */
+    function deleteBlacklistType(bytes32 _name) public withPerm(ADMIN) {
+        require(blacklists[_name].endTime != 0, "Blacklist type doesn’t exist");
+        require(blacklistToInvestor[_name].length == 0, "Investors are associated with the blacklist");
+        // delete blacklist type 
+        delete(blacklists[_name]);
+        emit DeleteBlacklistType(_name);
+    }
+
+    /**
+    * @notice Used to assign the blacklist type to the investor
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function addInvestorToBlacklist(address _investor, bytes32 _blacklistName) public withPerm(ADMIN) {
+        require(blacklists[_blacklistName].endTime != 0, "Blacklist type doesn't exist");
+        require(_investor != address(0), "Invalid investor address");
+        uint256 investorIndex = investorToBlacklist[_investor].length;
+        // Add blacklist index to the investor 
+        investorToIndex[_investor][_blacklistName] = investorIndex;
+        uint256 blacklistIndex = blacklistToInvestor[_blacklistName].length;
+        // Add investor index to the blacklist
+        blacklistToIndex[_blacklistName][_investor] = blacklistIndex;
+        investorToBlacklist[_investor].push(_blacklistName);
+        blacklistToInvestor[_blacklistName].push(_investor);
+        emit AddInvestorToBlacklist(_investor, _blacklistName);
+    }
+
+    /**
+    * @notice Used to delete the investor from the all associated blacklists
+    * @param _investor address of the investor
+    */
+    function deleteInvestorFromAllBlacklist(address _investor) public withPerm(ADMIN) {
+        require(_investor != address(0), "Invalid investor address");
+        require(investorToBlacklist[_investor].length != 0, "Investor is not associated to any blacklist type");
+        for(uint256 i = 0; i < investorToBlacklist[_investor].length; i++){
+            deleteInvestorFromBlacklist(_investor, investorToBlacklist[_investor][i]);
+        }
+    }
+
+    /**
+    * @notice Used to delete the investor from the blacklist
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function deleteInvestorFromBlacklist(address _investor, bytes32 _blacklistName) public withPerm(ADMIN) {
+        require(_investor != address(0), "Invalid investor address");
```

Comment:
@vpanwar821 could you add some check for the existence of the address. otherwise, it will run again for the deleted investor also.